sklearn/ensemble/gradient_boosting.py

```diff
@@ -2041,6 +2041,33 @@ class GradientBoostingClassifier(ClassifierMixin, BaseGradientBoosting):
     split. To obtain a deterministic behaviour during fitting,
     ``random_state`` has to be fixed.
 
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from sklearn.datasets import make_classification
+    >>> from sklearn.ensemble import GradientBoostingClassifier
+    >>> from sklearn.model_selection import train_test_split
+    >>> X, y = make_classification(n_samples=20, random_state=22)
+    >>> X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=5,
+    ...                                                     random_state=22)
+    >>> clf = GradientBoostingClassifier(n_estimators=2, max_depth=1,
```

Comment:
Why not just use the default n_estimators and max_depth? 2 and 1 aren't good values in general

Reply:
I chose these values mainly to demonstrate warm starting and staged prediction. With more estimators the output of staged_predict would have been very confusing.
If we keep the example simpler (which I will do based on your suggestion) then of course these values make no good sense. 
By the way is there an example about warm starting in general? I did not find one googling. So if not it may be worth to add one.

Reply:
There is one in the UG but it's quite hard to find I would agree https://scikit-learn.org/dev/modules/ensemble.html#fitting-additional-weak-learners